sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -447,11 +449,40 @@ public PipelineResponse(string requestId, HttpResponseMessage responseMessage, S
                 {
                     // Make sure we don't dispose the content if the stream was replaced
                     _responseMessage.Content = null;
+                    // Make sure we drop our cached value for the content
+                    _content = null;
 
                     _contentStream = value;
                 }
             }
 
+            public override BinaryData Content
+            {
+                get
+                {
+                    if (_content != null)
+                    {
+                        return _content;
+                    }
+
+                    if (ContentStream == null)
+                    {
+                        _content = new BinaryData(Array.Empty<byte>());
+                        return _content;
+                    }
+
+                    MemoryStream? contentMemoryStream = ContentStream as MemoryStream;
+                    if (contentMemoryStream == null)
+                    {
+                        contentMemoryStream = new MemoryStream();
+                        ContentStream.CopyTo(contentMemoryStream);
```

Comment:
I don't think we want to do it for very large and non seekable streams, e.g. the network stream. I think we should either throw for non-buffered streams, or check if the stream is seekable and has small length first.

Reply:
Also, I wonder if we should not change the buffered policy to work as follows: it buffers to BinaryData and stores it in response. The stream property is lazy and returns a stream wrapping the BD. The main reason for this is that is more optimal for our JsonDocument.Parse method (i.e. more efficient to parse Memory<byte>) [I think]